pydantic_ai_slim/pydantic_ai/durable_exec/temporal/_model.py

```diff
@@ -190,3 +235,92 @@ async def request_stream(
     def _validate_model_request_parameters(self, model_request_parameters: ModelRequestParameters) -> None:
         if model_request_parameters.allow_image_output:
             raise UserError('Image output is not supported with Temporal because of the 2MB payload size limit.')
+
+    def _get_model_id(self, model: models.Model | models.KnownModelName | str | None = None) -> str | None:
+        """Get the model ID for the given model parameter.
+
+        Returns a string that will be checked against registered model IDs,
+        or passed to infer_model if not found. Returns None to use the default model.
+        """
+        if model in (None, 'default'):
+            return None
+
+        if isinstance(model, Model):
+            # Check if this model instance is already registered
+            model_id = next((model_id for model_id, m in self._models_by_id.items() if m is model), ...)
+            if model_id is ...:
+                raise UserError(
+                    'Arbitrary model instances cannot be used at runtime inside a Temporal workflow. '
+                    'Register the model via `models` or reference a registered model by name.'
+                )
+            return None if model_id == 'default' else model_id
+
+        return model
+
+    def resolve_model(self, model: models.Model | models.KnownModelName | str | None = None) -> Model:
+        """Resolve a model parameter to a Model instance.
+
+        This is typically used outside of a workflow to resolve model parameters
+        before passing them to the underlying agent methods.
+
+        Args:
+            model: The model to resolve. Can be a Model instance, model name string,
+                   or None for the default model.
+
+        Returns:
+            The resolved Model instance.
+        """
+        # Handle Model instances directly - outside a workflow, unregistered
+        # Model instances are allowed since there's no serialization constraint.
+        if isinstance(model, Model):
+            # Check if registered and return it
+            for m in self._models_by_id.values():
+                if m is model:
+                    return m
+            # Not registered, return as-is for outside workflow use
+            return model
```

Comment:
The if above basically does nothing right? `if m is model: return m` is equivalent to `return model`, so just `if isinstance(model, Model): return model`? Maybe with an `in_workflow` check?